src/UsgsAstroLsSensorModel.cpp

```diff
@@ -704,112 +702,159 @@ csm::ImageCoord UsgsAstroLsSensorModel::groundToImage(
+     if (++count > 15) {
+       if (warnings) {
+         warnings->push_back(
+           csm::Warning(
+             csm::Warning::PRECISION_NOT_MET,
+             "Unable to achieve desired precision within 15 iterations.",
+             "UsgsAstroLsSensorModel::groundToImage()"));
+       }
+       break;
+     }
+     timei = getImageTime(approxPt);
+     detectorView = computeDetectorView(timei, groundPt, adj, desiredPrecision);
+
+     // Convert to detector line and sample
+     detectorLine = m_iTransL[0]
+                  + m_iTransL[1] * detectorView[0];
+                  + m_iTransL[2] * detectorView[1];
+
+     // Convert to image sample line
+     approxPt.line += detectorLine;
    }
 
-   // The computed pixel is the detector pixel, so we need to convert that to image lines
-   csm::ImageCoord calculatedPixel(detectorCoord.line + closestLine, detectorCoord.samp);
+   timei = getImageTime(approxPt);
+   detectorView = computeDetectorView(timei, groundPt, adj, desiredPrecision);
 
-   // Reintersect to ensure the image point actually views the ground point.
-   csm::EcefCoord calculatedPoint = imageToGround(calculatedPixel, height);
-   double dx = ground_pt.x - calculatedPoint.x;
-   double dy = ground_pt.y - calculatedPoint.y;
-   double dz = ground_pt.z - calculatedPoint.z;
-   double len = dx * dx + dy * dy + dz * dz;
+   // Invert distortion
+   double distortedFocalX, distortedFocalY;
+   applyDistortion(detectorView[0], detectorView[1], distortedFocalX, distortedFocalY,
+                   m_opticalDistCoeffs, m_distortionType, desiredPrecision);
 
-   if (achieved_precision) {
-      *achieved_precision = sqrt(len);
-   }
+   // Convert to detector line and sample
+   detectorLine = m_iTransL[0]
+                + m_iTransL[1] * distortedFocalX
+                + m_iTransL[2] * distortedFocalY;
+   detectorSample = m_iTransS[0]
+                  + m_iTransS[1] * distortedFocalX
+                  + m_iTransS[2] * distortedFocalY;
+
+   // Convert to image sample line
+   approxPt.line += detectorLine;
+   approxPt.samp = (detectorSample + m_detectorSampleOrigin - m_startingSample)
+                 / m_detectorSampleSumming;
+   achievedPrecision = &detectorLine;
+   MESSAGE_LOG(m_logger, "computeViewingPixel: image line sample {} {}",
+                                approxPt.line, approxPt.samp)
 
-   double preSquare = desired_precision * desired_precision;
-   if (warnings && (desired_precision > 0.0) && (preSquare < len)) {
-     MESSAGE_LOG(m_logger, "groundToImage: Desired precision not achieved {}", preSquare)
-     std::stringstream msg;
-     msg << "Desired precision not achieved. ";
-     msg << len << "  " << preSquare << "\n";
-     warnings->push_back(csm::Warning(csm::Warning::PRECISION_NOT_MET,
-                         msg.str().c_str(),
-                         "UsgsAstroLsSensorModel::groundToImage()"));
+   if (warnings && (desiredPrecision > 0.0) && (*achievedPrecision > desiredPrecision))

```

Comment:
Same here, needs a check that `achievedPrecision` is not NULL. You could use short-circuiting here by just adding `&& (achievedPrecision)` in the middle of these two.